extras/kinesis-asl/src/test/scala/org/apache/spark/streaming/kinesis/KinesisSourceSuite.scala

```diff
@@ -17,63 +17,215 @@
+        .option("endpoint", testUtils.endpointUrl)
+        .kinesis().stream()
+      sqlContext.read
+        .option("stream", testUtils.streamName)
+        .option("region", testUtils.regionName)
+        .kinesis().stream()
+
+      assertExceptionAndMessage[IllegalArgumentException](
+        "Either option 'region' or option 'endpoint' must be specified. Examples: " +
+          """option("region", "us-west-2"), """ +
+          """option("endpoint", "https://kinesis.us-west-2.amazonaws.com")""") {
+        sqlContext.read.option("stream", testUtils.streamName).kinesis().stream()
+      }
+      assertExceptionAndMessage[IllegalArgumentException](
+        s"'region'(invalid-region) doesn't match to 'endpoint'(${testUtils.endpointUrl})") {
+        sqlContext.read
+          .option("stream", testUtils.streamName)
+          .option("region", "invalid-region")
+          .option("endpoint", testUtils.endpointUrl)
+          .kinesis().stream()
+      }
+
+      // position
+      assertExceptionAndMessage[IllegalArgumentException](
+        "Unknown value of option 'position': invalid") {
+        sqlContext.read
+          .option("stream", testUtils.streamName)
+          .option("endpoint", testUtils.endpointUrl)
+          .option("position", "invalid")
+          .kinesis().stream()
+      }
+
+      // accessKey and secretKey
+      assertExceptionAndMessage[IllegalArgumentException](
+        "'accessKey' is set but 'secretKey' is not found") {
+        sqlContext.read
+          .option("stream", testUtils.streamName)
+          .option("endpoint", testUtils.endpointUrl)
+          .option("position", InitialPositionInStream.TRIM_HORIZON.name())
+          .option("accessKey", "test")
+          .kinesis().stream()
+      }
+      assertExceptionAndMessage[IllegalArgumentException](
+        "'secretKey' is set but 'accessKey' is not found") {
+        sqlContext.read
+          .option("stream", testUtils.streamName)
+          .option("endpoint", testUtils.endpointUrl)
+          .option("position", InitialPositionInStream.TRIM_HORIZON.name())
+          .option("secretKey", "test")
+          .kinesis().stream()
+      }
+    } finally {
       testUtils.deleteStream()
     }
-    super.afterAll()
   }
 
-  test("basic receiving") {
-    val kinesisSource = KinesisSource(
-      sqlContext,
-      testUtils.regionName,
-      testUtils.endpointUrl,
-      Set(streamName),
-      InitialPositionInStream.TRIM_HORIZON)
-    val mapped = kinesisSource.toDS().map[Int]((bytes: Array[Byte]) => new String(bytes).toInt + 1)
-    val testData = 1 to 10 // This ensures that data is sent to multiple shards for 2 shard streams
-    testStream(mapped)(
-      AddKinesisData(kinesisSource, testData: _*),
-      CheckAnswer(testData.map { _ + 1 }: _*)
-    )
+  testIfEnabled("call kinesis when not using stream") {
+    val expectedMessage = "org.apache.spark.streaming.kinesis.DefaultSource is " +
```

Comment:
since this message is generated by some other component other than kinesis, if they change message this function could silently break (as kinesis tests dont run always). So dont check the message.